src/server/common_services/auth_server.js

```diff
@@ -669,9 +669,8 @@ function _get_auth_info(account, system, authorized_by, role, extra) {
  * @param {string} bucket_path s3 bucket path (must start from "/")
  * @returns  {Promise<boolean>} true if the account has permission to perform the action on the bucket
  */
-async function has_bucket_action_permission(bucket, account, action, bucket_path = "") {
+async function has_bucket_action_permission(bucket, account, action, req_query, bucket_path = "") {
```

Comment:
Something looks weird - in the function signature the argument order is: `bucket, account, action, req_query, bucket_path` (`req_query` and then `bucket_path`)
But when you edited the calls, it is (for example): `return req.has_bucket_action_permission(bucket, action, bucket_path, req_query);` (`bucket_path` and then `req_query`)

notice that by changing the function signature you need to update the JSDoc.

Reply:
Yes, it is confusing...I also took time to understand what is going on...
but req.has_s3_bucket_permission  is different from has_s3_bucket_permission 

req.has_s3_bucket_permission = async function(bucket, action, bucket_path, req_query) {
and 
async function has_bucket_action_permission(bucket, account, action, req_query, bucket_path = "") {

I wish names were different.